From 722f7679bece51e4d3da8a53630e7bb36ea56b76 Mon Sep 17 00:00:00 2001
From: Kirill Kapranov <kirill.kapranov@compulab.co.il>
Date: Tue, 26 May 2020 21:17:29 +0300
Subject: [PATCH 07/49] compulab: integrate DM_I2C in EEPROM driver

CompuLab's EEPROM driver is enabled only for CONFIG_SYS_I2C.
Update the  enabling condition of the driver to CONFIG_SYS_I2C or
CONFIG_DM_I2C.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 board/compulab/common/Makefile | 4 +++-
 board/compulab/common/eeprom.h | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/board/compulab/common/Makefile b/board/compulab/common/Makefile
index 1a8b63fb4b..500858035f 100644
--- a/board/compulab/common/Makefile
+++ b/board/compulab/common/Makefile
@@ -6,6 +6,8 @@
 
 obj-y				+= common.o
 obj-y				+= mmc.o
-obj-$(CONFIG_SYS_I2C)		+= eeprom.o
+ifneq "$(or $(CONFIG_SYS_I2C),$(CONFIG_DM_I2C))" ""
+obj-y				+= eeprom.o
+endif # (CONFIG_SYS_I2C || CONFIG_DM_I2C2)
 obj-$(CONFIG_LCD)		+= omap3_display.o
 obj-$(CONFIG_SMC911X)		+= omap3_smc911x.o
diff --git a/board/compulab/common/eeprom.h b/board/compulab/common/eeprom.h
index a9c0203b81..2628d4ab86 100644
--- a/board/compulab/common/eeprom.h
+++ b/board/compulab/common/eeprom.h
@@ -10,7 +10,7 @@
 #define _EEPROM_
 #include <errno.h>
 
-#ifdef CONFIG_SYS_I2C
+#if (defined(CONFIG_SYS_I2C) || defined(CONFIG_DM_I2C))
 int cl_eeprom_read_mac_addr(uchar *buf, uint eeprom_bus);
 u32 cl_eeprom_get_board_rev(uint eeprom_bus);
 int cl_eeprom_get_product_name(uchar *buf, uint eeprom_bus);
-- 
2.11.0

